roles/slurm/templates/notification_scripts/nodes.bash

```diff
@@ -1,23 +1,146 @@
+		\`\`\`"
+	}\0
 EOM
 
+#logger --id="${$}" -s "Got new node state stored in ${node_state_file_new}."
+
+#
+# Delete old node state after one day to send a new notification
+# if the problem was not yet resolved.
+#
+if [[ -e "${node_state_file_old}" ]]; then
+	find "${node_state_file_old}" -mtime +1 -delete
+fi
+
+#
+# Compile new notification.
+#
+if [[ -e "${node_state_file_old}" ]]; then
+	if [[ -r "${node_state_file_old}" ]]; then
+		#logger --id="${$}" -s "${node_state_file_old} exists and is readable."
+		#
+		# Drop some resolution from the "REASON" field to prevent sending redundant notifications
+		# when the state has not changed significantly.
+		# E.g. load was too high and changed slightly, but is still is too high.
+		#
+		node_state_file_old_short="${node_state_file_old}-${$}.short"
+		node_state_file_new_short="${node_state_file_new}.short"
+		sed 's/:  [^:]*$//' "${node_state_file_old}" | sort > "${node_state_file_old_short}"
+		sed 's/:  [^:]*$//' "${node_state_file_new}" | sort > "${node_state_file_new_short}"
+		#logger --id="${$}" -s "Created ${node_state_file_old_short} and ${node_state_file_new_short}."
+		if cmp -s "${node_state_file_old_short}" "${node_state_file_new_short}"; then
+			: # no-op: The files are identical: nothing changed and there is no need to send a new notification.
+			#logger --id="${$}" -s "No differences found in ${node_state_file_old_short} compared to ${node_state_file_new_short}."
+		else
+			#logger --id="${$}" -s "Differences were found in ${node_state_file_old_short} compared to ${node_state_file_new_short}."
+			cp -v "${node_state_file_new}" "${node_state_file_old}"
+			send_notification='true'
+			#logger --id="${$}" -s "send_notification = ${send_notification}."
+		fi
+		rm -v -f "${node_state_file_old_short}" "${node_state_file_new_short}"
+	else
+		#
+		# Compile JSON error message payload.
+		#
+		read -r -d '\0' message <<- EOM
+			{
+				"type": "mrkdwn",
+				"text": "*Failed to report node state for _{{ slurm_cluster_name | capitalize }}_*:  
+				${node_state_file_old} exists but is not readable."
+			}\0
+		EOM
+		send_notification='true'
+		#logger --id="${$}" -s "send_notification = ${send_notification}."
+	fi
+else
+	#logger --id="${$}" -s "${node_state_file_old} does not exist: send notification."
+	cp -v "${node_state_file_new}" "${node_state_file_old}"
+	send_notification='true'
+	#logger --id="${$}" -s "send_notification = ${send_notification}."
+fi
+
 #
 # Post message to Slack channel.
 #
-curl -X POST '{{ slurm_notification_slack_webhook }}' \
-	 -H 'Content-Type: application/json' \
-	 -d "${message}"
+if [[ "${send_notification}" == 'true' ]]; then
+	#logger --id="${$}" -s "Sending message..."
+	curl -X POST '{{ slurm_notification_slack_webhook }}' \
+		 -H 'Content-Type: application/json' \
+		 -d "${message}"
+else
+	: # no-op
+	#logger --id="${$}" -s 'Node state has not changed: not sending message.'
+fi
+
+#
+# Cleanup
+#
+rm -v -f "${node_state_file_new}"
```

Comment:
If I see correct, this is the only place that cleans old state files with specific `${$}`.

So if if happens that the script crashes while running, this line is not reached and old state files pile up for every time that this happens.

I would add a few lines in the front which clean up any leftover of old files
`node_state_file_new="${node_state_dir}/sinfo_report_new-${$}"`

Reply:
Point taken; I'll implement signal trapping and a cleanup routine...